sql/core/src/main/scala/org/apache/spark/sql/Column.scala

```diff
@@ -1222,6 +1222,43 @@ class Column(val expr: Expression) extends Logging {
    */
   def cast(to: String): Column = cast(CatalystSqlParser.parseDataType(to))
 
+  /**
+   * Casts the column to a different data type and the result is null on failure.
+   * {{{
+   *   // Casts colA to IntegerType.
+   *   import org.apache.spark.sql.types.IntegerType
+   *   df.select(df("colA").try_cast(IntegerType))
+   *
+   *   // equivalent to
+   *   df.select(df("colA").try_cast("int"))
+   * }}}
+   *
+   * @group expr_ops
+   * @since 4.0.0
+   */
+  def try_cast(to: DataType): Column = withExpr {
+    val cast = Cast(
+      child = expr,
+      dataType = CharVarcharUtils.replaceCharVarcharWithStringForCast(to),
+      evalMode = EvalMode.TRY)
+    cast.setTagValue(Cast.USER_SPECIFIED_CAST, ())
```

Comment:
Just curious, what is this tag doing?

Reply:
https://github.com/apache/spark/blob/f18d945af7b69fbc89b38b9ca3ca79263b0881ed/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala#L182-L185

I guess it is used to distinguish between user calls and internal calls.
Seems we never support it in Spark Connect though

Reply:
this tag was introduced in https://github.com/apache/spark/commit/92e5cfd58dc7dd5a5b1d3e2b6e93a03af5a07b67, seems it affects the `sql` string and `prettyName`

Reply:
it seems this tag was used to change the field name in sql text `spark.sql("select floor(1)")`

I suspect it doesn't take affect in dataframe API, I compared the output column name of `cast/try_cast`, and the classic and connect are the same.